internal/k8s/k8s.go

```diff
@@ -297,6 +300,25 @@ func New(cfg *Config) (*Client, error) {
 		}
 	}
 
+	if cfg.BGPStatusChan != nil {
```

Comment:
When that can take place?

Reply:
I mean when the cfg.BGPStatusChan is nil

Reply:
when the "controller" pod (the one that is in charge of allocating ips) calls k8s.New